File: java/src/main/java/de/thorbenkuck/kafka/KafkaConsumerConfiguration.java
package de.thorbenkuck.kafka;

import de.thorbenkuck.kafka.data.ExampleMessage;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;

@Configuration
public class KafkaConsumerConfiguration {

    private final SmartContainerFactory factory;

    public KafkaConsumerConfiguration(SmartContainerFactory factory) {
        this.factory = factory;
    }

    @Bean("exampleMessageContainerFactory")
    public ConcurrentKafkaListenerContainerFactory<String, ExampleMessage> exampleMessageContainerFactory(
            KafkaConsumerProperties properties
    ) {
        return factory.createContainerFactory(properties.getTopic(), ExampleMessage.class);
    }


}

File: kotlin/src/main/kotlin/de/thorbenkuck/kafka/KafkaConsumerConfiguration.kt
package de.thorbenkuck.kafka

import de.thorbenkuck.kafka.data.ExampleMessage
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory

@Configuration
class KafkaConsumerConfiguration(
		private val factory: SmartContainerFactory
) {

	@Bean("exampleMessageContainerFactory")
	fun exampleMessageContainerFactory(
			properties: KafkaConsumerProperties
	): ConcurrentKafkaListenerContainerFactory<String, ExampleMessage> = factory.createContainerFactory(properties.topic)

}